aiaccel/hpo/samplers/nelder_mead_sampler.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import annotations
+
+import queue
+import warnings
+from collections.abc import Generator
+from dataclasses import dataclass
+from typing import Any, Sequence
+
+import numpy as np
+import optuna
+from optuna.distributions import BaseDistribution
+from optuna.study import Study
+from optuna.trial import FrozenTrial, TrialState
+
+
+@dataclass
+class NelderMeadCoefficient:
+    r: float = 1.0
+    ic: float = -0.5
+    oc: float = 0.5
+    e: float = 2.0
+    s: float = 0.5
+
+
+class NelderMeadAlgorism:
+    vertices: np.ndarray[Any, Any]
+    values: np.ndarray[Any, Any]
+
+    def __init__(
+        self,
+        search_space: dict[str, list[float]],
+        coeff: NelderMeadCoefficient | None = None,
+        seed: int | None = None,
+    ) -> None:
+        self._search_space = search_space
+        self.coeff = coeff if coeff is not None else NelderMeadCoefficient()
+
+        self.dimension = len(search_space)
+
+        self.value_queue: queue.Queue[float] = queue.Queue()
+        self._rng: np.random.RandomState = np.random.RandomState(seed)
+        self.is_ready: bool = False
```

Comment:
右辺の型から型推論できると思うので，型アノテーションは不要な気がしましたが如何でしょう？
```suggestion
        self._rng = np.random.RandomState(seed)
        self.is_ready = False
```

Reply:
反映しました。

01347c49aa643c387d322b40ce8f75dc337857e1